### DISCUSSION TITLE
Bookable entities

### DISCUSSION THREAD

**Discussion by dawidurbanski** [wpappointments/wpappointments#89] [Ideas]
Make it possible to make (possibly nested), bookable entities.

Bookable entity would be a thing that can have it's own availability schedule, could have specified list of available services, assigned employees and could be added to one or many locations.

**Comment by wppoland** [comment]
## Implementation Progress

The Bookable Entities backend data model has been implemented in [`110a69c`](https://github.com/wpappointments/wpappointments/commit/110a69c).

### What's done

- **Hierarchical CPT** `wpa-entity` — supports nesting via `post_parent` (e.g. restaurant → table → seat)
- **Entity Model** — full CRUD with schedule inheritance (`inherit`/`own` mode), buffer/lead time inheritance walking up the hierarchy to global defaults, capacity, type, active/inactive
- **Entity Query** — `all()`, `active()`, `roots()`, `children()`, `tree()` (recursive nested builder), `ancestors()`, filtering by type/parent/active
- **REST API** — `GET/POST /entities`, `GET/PUT/DELETE /entities/{id}`, `GET /entities/tree`, `GET /entities/{id}/children`, `GET/PUT /entities/{id}/services`
- **Service linking** — many-to-many relationship via `service_ids`/`entity_ids` post meta with bidirectional sync
- **Capability** — `wpa_manage_entities` added and registered
- **TypeScript** — Entity schema, types, and full API layer

### Still needed

- Admin UI for managing entities (list, create, edit, assign services)
- Entity selection step in booking flow
- Migration path from current flat appointment model
- Employees and Locations — premium addons (scaffolding deployed to `wpappointments-premium`)

### Architecture decisions

- **Schedule inheritance:** entity → parent entity → … → global default. Most specific wins.
- **Buffer/lead times:** same hierarchical override pattern.
- **Nesting:** unlimited depth, children reassigned to parent on delete.
- **Services:** many-to-many, each entity can offer different services.
